File: 2020-10-30-client-libraries/ruby/authentication.rb
require 'stripe'

# Globally set API key:
# Stripe.api_key = 'sk_test...'

# API key set per request:
# customer_id = 'cus_ICiIH7WIPI4Gr1'
# customer = Stripe::Customer.retrieve(customer_id, {api_key: 'sk_test...'})
# p customer

# Authenticating API calls with connect
account_id = 'acct_1Ey3h1BqeQ4DKpna'
customer_id = 'cus_HDfWzCQ6UEVtfu'

# Setting your platform secret API key globally:
# Stripe.api_key = 'sk_test...'
# p Stripe::Customer.retrieve(customer_id, {
#   stripe_account: account_id,
# })

# Or per request:
p Stripe::Customer.retrieve(customer_id, {
  stripe_account: account_id,
  api_key: 'sk_test...'
})

File: 2020-10-30-client-libraries/python/authentication.py
import stripe

# Set globally

# stripe.api_key = "sk_test..."

# Per request
# print(stripe.Customer.retrieve('cus_ICiIH7WIPI4Gr1', api_key='sk_test...'))

# With connect
account_id = 'acct_1Ey3h1BqeQ4DKpna'

# Per request
# print(stripe.Customer.list(stripe_account=account_id, api_key='sk_test...'))

# Set platform API key globally
stripe.api_key = "sk_test..."
print(stripe.Customer.retrieve('cus_HDfWzCQ6UEVtfu', stripe_account=account_id))
